test/ProblemDetailsAssertionsTrait.php

```diff
@@ -84,39 +83,41 @@ public function prepareResponsePayloadAssertions(
     }
 
     /**
-     * @param StreamInterface|ObjectProphecy $stream
+     * @param StreamInterface|MockObject $stream
      */
-    public function preparePayloadForJsonResponse(ObjectProphecy $stream, callable $assertion) : void
+    public function preparePayloadForJsonResponse(MockObject $stream, callable $assertion): void
     {
         $stream
-            ->write(Argument::that(function ($body) use ($assertion) {
-                Assert::assertInternalType('string', $body);
+            ->expects($this->any())
+            ->method('write')
+            ->with($this->callback(function ($body) use ($assertion) {
+                Assert::assertIsString($body);
                 $data = json_decode($body, true);
                 $assertion($data);
-                return $body;
-            }))
-            ->shouldBeCalled();
+                return true;
```

Comment:
I have changed the return type of callbacks in a few places.

PHPUnit actually expects them to return a boolean, making the test fail if the value is `false`.

In previous versions there was an implicit casting, so returning any truthy value was considered as true. However, PHPUnit now has declared the return type as `bool`, making the tests fail because they were returning something else.

Since all of them are doing assertios before the return, I have changed the returns to simply `return true`.